staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -71,22 +71,59 @@ type EgressSelection struct {
 
 // Connection provides the configuration for a single egress selection client.
 type Connection struct {
-	// type is the type of connection used to connect from client to network/konnectivity server.
-	// Currently supported values are "http-connect" and "direct".
-	Type string `json:"type"`
+	// Protocol is the protocol used to connect from client to the konnectivity server.
+	ProxyProtocol ProtocolType `json:"proxyProtocol,omitempty"`
 
-	// httpConnect is the config needed to use http-connect to the konnectivity server.
-	// Absence when the type is "http-connect" will cause an error
-	// Presence when the type is "direct" will also cause an error
+	// Transport defines the transport configurations we use to dial to the konnectivity server
 	// +optional
-	HTTPConnect *HTTPConnectConfig `json:"httpConnect,omitempty"`
+	Transport *Transport `json:"transport,omitempty"`
 }
 
-type HTTPConnectConfig struct {
-	// url is the location of the proxy server to connect to.
+// ProtocolType is a set of valid values for Connection.ProtocolType
+type ProtocolType string
+
+// Valid types for ProtocolType for konnectivity server
+const (
+	// Use http-connect to connect to konnectivity server
+	ProtocolHTTPConnect ProtocolType = "http-connect"
+	// Use grpc to connect to konnectivity server
+	ProtocolGRPC ProtocolType = "grpc"
+	// Connect directly (skip konnectivity server)
+	ProtocolDirect ProtocolType = "direct"
+)
+
+// Transport defines the transport configurations we use to dial to the konnectivity server
+type Transport struct {
+	// TCP is the TCP configuration for communicating with the konnectivity server via TCP
+	// Field is optional but one of TCP or UDS must be present if Protocol is http-connect or GRPC
+	// +optional
+	TCP *TCPTransport `json:"tcp,omitempty"`
+
+	// UDS is the UDS configuration for communicating with the konnectivity server via UDS
+	// Field is optional but one of TCP or UDS must be present if Protocol is http-connect or GRPC
+	// +optional
+	UDS *UDSTransport `json:"uds,omitempty"`
```

Comment:
would it be simpler to:
* require the *Transport field be nil for Direct, and non-nil for HTTPConnect/GRPC
* for non-nil *Transport object, require exactly one of TCP/UDS be set?